python/semantic_kernel/functions/kernel_function_extension.py

```diff
@@ -23,11 +23,25 @@
     )
     from semantic_kernel.functions.kernel_function import KernelFunction
     from semantic_kernel.functions.types import KERNEL_FUNCTION_TYPE
+    from semantic_kernel.kernel import Kernel
 
 
 logger: logging.Logger = logging.getLogger(__name__)
 
 
+@runtime_checkable
+class AddToKernelCallbackProtocol(Protocol):
+    """Protocol for the callback to be called when the plugin is added to the kernel."""
+
+    def added_to_kernel(self, kernel: "Kernel") -> None:
```

Comment:
Can you help me understand the utility of this? What actions would I want to take once a plugin is added to the kernel?

Reply:
For this specific instance, I need the kernel to be able to do `sampling` (more accurately I need `services`) but I figured there might be other instances, maybe in the agent space, where after a Agent is added to the kernel, you want to run some logic that alters the behavior...